sources/train.py

```diff
@@ -150,27 +154,39 @@
                             "'medium' values"))
         sys.exit(1)
 
-    # Instance name (name + image size + network size)
-    instance_name = args.name + "_" + str(args.image_size) + "_" + args.network_size
+    if not args.model in ["feature_detection", "semantic_segmentation"]:
+        utils.logger.error("Unsupported model. Please consider ")
+        utils.logger.utils(("'feature_detection' or 'semantic_segmentation'"))
+        sys.exit(1)
+
     # Data path and repository management
     dataset_repo = os.path.join(args.datapath, args.dataset)
-    suffix = "" if not args.aggregate_label else "aggregate_"
-    training_name = "training_" + suffix + str(args.image_size)
-    validation_name = "validation_" + suffix + str(args.image_size)
-    os.makedirs(os.path.join(dataset_repo, training_name, 'images'), exist_ok=True)
-    os.makedirs(os.path.join(dataset_repo, training_name, 'labels'), exist_ok=True)
-    os.makedirs(os.path.join(dataset_repo, validation_name, 'images'), exist_ok=True)
-    os.makedirs(os.path.join(dataset_repo, validation_name, 'labels'), exist_ok=True)
-    training_filename = os.path.join(dataset_repo, training_name + '.json')
-    validation_filename = os.path.join(dataset_repo, validation_name + '.json')
+    input_repo = os.path.join(dataset_repo, "input")
+    aggregate_value = "full" if not args.aggregate_label else "aggregate"
+    preprocessed_repo = str(args.image_size) + "_" + aggregate_value
+    preprocessed_path = os.path.join(dataset_repo, "preprocessed", preprocessed_repo)
+    training_filename = os.path.join(preprocessed_path, "training.json")
+    validation_filename = os.path.join(preprocessed_path, "validation.json")
+    preprocessed_training_path = os.path.join(preprocessed_path, "training")
+    preprocessed_validation_path = os.path.join(preprocessed_path, "validation")
+    backup_path = os.path.join(dataset_repo, "output", args.model)
+    os.makedirs(os.path.join(preprocessed_training_path, "images"), exist_ok=True)
+    os.makedirs(os.path.join(preprocessed_training_path, "labels"), exist_ok=True)
+    os.makedirs(os.path.join(preprocessed_validation_path, "images"), exist_ok=True)
+    os.makedirs(os.path.join(preprocessed_validation_path, "labels"), exist_ok=True)
+    os.makedirs(backup_path, exist_ok=True)
+
+    # Instance name (name + image size + network size + batch_size + aggregate? + dropout + learning_rate)
+    instance_name = (args.name + "_" + str(args.image_size) + "_" + args.network_size
```

Comment:
Maybe you can put all these variables in a list and call the `list_to_str` function with a `_` separator

Reply:
Could be possible. Not sure it is a fundamental improvement, however it would be cleaner, for sure.

Reply:
Agreed.